src/odemis/driver/smaract.py

```diff
@@ -3173,3 +3174,1239 @@ def SA_CTL_Move(self, handle, ch, pos_pm, _):
+        self.core.SA_SI_GetValue_f64(self._id, channel, data_source_idx, byref(ret_val))
+        return ret_val.value
+
+    def WaitForEvent(self, timeout=600):
+        """
+        Blocks until device reports an event.
+        timeout (int > 0): timeout in seconds
+        returns (SA_SI_Event): event
+        raises
+            TimeoutError: timeout exceeded
+            CancelledError: function was cancelled by SA_SI_Cancel
+            SA_SI_Error: something went wrong
+        """
+        if timeout == float("inf"):
+            t = SA_SI_TIMEOUT_INFINITE
+        else:
+            t = c_uint(int(timeout * 1000))  # SA_SI_WaitForEvent accepts timeout in ms
+
+        ev = SA_SI_Event()
+        ret = self.core.SA_SI_WaitForEvent(self._id, byref(ev), t)
+
+        if ret != SA_SI_ERROR_NONE:
+            if ret == SA_SI_ERROR_TIMEOUT:
+                raise TimeoutError("Picoscale reported timeout error.")
+            elif ret == SA_SI_ERROR_CANCELLED:
+                raise CancelledError("WaitForEvent was cancelled.")
+            else:
+                if ret in self.core.err_code:
+                    raise SA_SIError(ret, self.core.err_code[ret])
+                else:
+                    raise SA_SIError(ret, "Unknown error code %s" % ret)
+        return ev
+
+    def _load_configuration(self):
+        """
+        Load the configuration of the device. The loaded parameters are
+        * fiber length
+        * extension fiber length
+        * number of active channels
+        * sensor head types
+        * minimum and maximum working distance
+        These configuration parameters need to be set during the installation of the system.
+        Additionally, the parameters from the previously saved referencing procedure are loaded.
+        This function is blocking.
+        """
+        # Loading the configuration will cause the state to become temporarily unstable. Once the process is
+        # finished, the state will be stable again.
+        self.EnableEventNotification(SA_PS_STABLE_STATE_CHANGED_EVENT)
+        self.SetProperty_i32(SA_PS_AF_ADJUSTMENT_RESULT_LOAD_PROP, SA_SI_ENABLED)
+        self._wait_for_progress_event(SA_SI_ENABLED, timeout=600)  # this can take a long time
+        self.DisableEventNotification(SA_PS_STABLE_STATE_CHANGED_EVENT)
+
+    def _save_configuration(self):
+        """
+        Saves the configuration parameters (cf _load_configuration) and internal parameters of the
+        referencing procedure. This function is blocking.
+        """
+        self.SetProperty_i32(SA_PS_AF_ADJUSTMENT_RESULT_SAVE_PROP, SA_SI_ENABLED)
+        # There is no documentation on loading/saving the configuration in the manual, even though
+        # the functions are available in the header file. It is not clear how much time saving the
+        # parameters takes. It can be assumed that it's fast --> wait 1 s.
+        time.sleep(1)
+
+    def _wait_for_progress_event(self, end_state, timeout=600):
```

Comment:
I'd suggest to put the default timeout to infinity, as you used to. 10min is really arbitrary, so it'd be much more readable to have it explicitly written in every place you call this function (and you want a 10 min timeout). Otherwise, it's really hard to guess when you read the caller's code that this function times out after 10 min.